tests/integration/test_charm.py

```diff
@@ -126,3 +126,20 @@ async def test_reset_instance_action(
     config = await model.applications[another_synapse_app.name].get_config()
     current_server_name = config.get("server_name", {}).get("value")
     assert current_server_name == another_server_name
+
+
+async def test_register_user_action(model: Model, synapse_app: Application) -> None:
+    """
+    arrange: a deployed Synapse charm.
+    act: call the register user action.
+    assert: the user is registered.
+    """
+    unit = model.applications[synapse_app.name].units[0]
+    action_register_user: Action = await synapse_app.units[0].run_action(  # type: ignore
+        "register-user", username="operator", admin="yes"
+    )
+    await action_register_user.wait()
+    assert action_register_user.status == "completed"
+    assert action_register_user.results["register-user"]
+    assert action_register_user.results["user-password"]
+    assert unit.workload_status == "active"
```

Comment:
There is a standard here which states that further checks should be done to ensure the action worked, is that applicable in this case? https://juju.is/docs/sdk/styleguide#heading--integration-tests

Reply:
Thanks, I added a login check.